### ISSUE TITLE
Add a notice that translation keys must be globally unique

### ISSUE THREAD

**Issue by fauxnik** [derail-valley-modding/language-helper#5]
It's not currently clear that translation keys must be globally unique. One could be led to believe by the existence of the `sourceId` passed to the translations injector that they are somehow namespaced automatically.

**Comment by Insprill** [comment]
Is there a reason they aren't namespaced automatically? I think that would be the better solution since creating globally unique keys that don't conflict with any other mods is going to be a never-ending goose chase.

**Comment by katycat5e** [comment]
Automatic prefixes introduce issues with any calls to Localize, and especially hardcoded keys in assets - you would need to know what the automatic prefix is in order to set up your keys correctly. I think the notice to modders to add their own prefix is likely the most straightforward solution.

**Comment by fauxnik** [comment]
Can you say more about the hardcoded keys in assets? Because it's totally possible that the helper could simply proxy the localize method and add the prefix automatically. I made a helper method in my own project to do just that. Depending on how csv/web transition injections are handled, I imagine it'd be possible to add an automatic prefix on the injection side too.

**Comment by katycat5e** [comment]
Injection side is easy. But in CCL, we're using base game components that take localization keys and use the built in localize methods. There's no way to differentiate between what keys should be prefixed or not if we were to try to patch the localize method. In other mods you'd have to remember to call the langhelper localize function instead of using the existing system.

**Comment by fauxnik** [comment]
Yeah, definitely no way to patch the localize method and get it right.

Having to remember to call the right method versus having to remember to prefix all mod keys isn't all that different imo, but the former involves less work for the developer and has a very obvious symptom when it's forgotten, i.e. `[missing translation]`
